### PR TITLE
Proper custom job name support

### PR DESCRIPTION
**Pull request by Tastyfish**
## About The Pull Request

Here at Plexagon Corp, we acknowledge that in these ever-changing times, there is more need than ever for custom job assignments. Thus, we made our systems finally support them properly.

* The datacore now has a hidden `trim` field that contains the trim assignment of the id card, or the same value as `rank` in all spawning and legacy situations.
* ID cards now have a `get_trim_assignment()` proc that better modularizes and breaks out the logic from `get_sechud_job_icon_state()` making the ID card code incidentally cleaner.
* When the HoP assigns a card using the Plexagon Access System program, `trim` is assigned to the id card's get_trim_assignment. This way, the "real" job underlying a custom job name is still maintained.
* The PDA manifest now uses `trim` to sort them, finally resolving the issue of custom jobs being mixed into Unassigned.
* The security console now shows the initial rank on spawn/latejoin (stored in the general datacore as `field["initial_rank"]`.
* The admin secrets manifest shows the trim in parenthesis, if it differs.
* To follow this pattern, the crew monitoring computer now also uses `trim` to color and sort crew instead of `assignment`.
  * Further, the crew monitor will now place completely unrecognized jobs at the bottom as it should have.

Manifest
Monitor
Datacore
Secrets

## Why It's Good For The Game

Custom jobs finally don't make all of the interfaces stain their pants.

## Changelog

:cl:
fix: The crew monitor and crew manifest now correctly sort custom jobs according to ID trim.
refactor: Data cores now internally track ID trim when HoP updates cards, as well as initial job.
admin: The secrets manifest now also shows the registered trim for each player.
/:cl:

### CODE REVIEW

**Comment by Tastyfish** [comment]
As far as I can tell, the linter failed but with no errors listed.

**Comment by Bawhoppen** [comment]
I think this is a good change, although I maybe would request that non-roundstart/true latejoin jobs get noted somehow? I could see this being of importance for some detective work or something potentially.

**Comment by Tastyfish** [comment]
Ultimately, their original roundstart/latejoin job is still stored in `mob.job`, so you can easily see if somebody changed their job at the HOP console, or did you mean tracking something else?

**Comment by Bawhoppen** [comment]
> Ultimately, their original roundstart/latejoin job is still stored in `mob.job`, so you can easily see if somebody changed their job at the HOP console, or did you mean tracking something else?

Oh I was not aware of that. I was just thinking from an ingame perspective for being able to gather that info.

**Comment by Tastyfish** [comment]
> 
> 
> > Ultimately, their original roundstart/latejoin job is still stored in `mob.job`, so you can easily see if somebody changed their job at the HOP console, or did you mean tracking something else?
> 
> Oh I was not aware of that. I was just thinking from an ingame perspective for being able to gather that info.

Hm, yeah currently there is no audit trail built into, say, the data core. I could add a field to the data core that notes their original job, is visible to the security console, but is read only, but of course would have to be unassigned for manually created records. Is that what you were thinking?

**Comment by Bawhoppen** [comment]
> > > Ultimately, their original roundstart/latejoin job is still stored in `mob.job`, so you can easily see if somebody changed their job at the HOP console, or did you mean tracking something else?
> > 
> > 
> > Oh I was not aware of that. I was just thinking from an ingame perspective for being able to gather that info.
> 
> Hm, yeah currently there is no audit trail built into, say, the data core. I could add a field to the data core that notes their original job, is visible to the security console, but is read only, but of course would have to be unassigned for manually created records. Is that what you were thinking?

Something like that would be appreciated by many I think. Definitely by me.

**Comment by Tastyfish** [comment]
> 
> 
> > > > Ultimately, their original roundstart/latejoin job is still stored in `mob.job`, so you can easily see if somebody changed their job at the HOP console, or did you mean tracking something else?
> > > 
> > > 
> > > Oh I was not aware of that. I was just thinking from an ingame perspective for being able to gather that info.
> > 
> > 
> > Hm, yeah currently there is no audit trail built into, say, the data core. I could add a field to the data core that notes their original job, is visible to the security console, but is read only, but of course would have to be unassigned for manually created records. Is that what you were thinking?
> 
> Something like that would be appreciated by many I think. Definitely by me.

I've added seeing the initial rank in the security console, as well as seeing trim in the admin secrets menu. See the initial PR description for pictures.

File: code/game/machinery/computer/security.dm [hunk 1]
```diff
@@ -919,8 +922,7 @@ What a mess.*/
 					if("Change Rank")
 						if(active1)
 							active1.fields["rank"] = strip_html(href_list["rank"])
-							if(href_list["rank"] in SSjob.station_jobs)
-								active1.fields["real_rank"] = href_list["real_rank"]
```
Comment (AnturK) [inline code/game/machinery/computer/security.dm:923]: Was this param unused ?

**Review by AnturK — APPROVED** [review]
Looks alright

File: code/game/machinery/computer/security.dm (hunk 1 above)
Comment (Tastyfish) [reply code/game/machinery/computer/security.dm:923]: Yes, this is the only instance of real_name used in the entire codebase. Just a cleanup.

**Pull request merged**